test/Prototypes/Algorithms.swift.gyb

```diff
@@ -43,21 +43,11 @@ public protocol MutableCollectionAlgorithms : MutableCollection {
   /// - Complexity: O(*n*)
   @discardableResult
   mutating func rotate(shiftingToStart middle: Index) -> Index
-
-  /// Rotates the elements in `bounds` so that the element
-  /// at `middle` ends up first in `bounds`.
-  ///
-  /// - Returns: The new index of the element that was first
-  ///   pre-rotation.
-  /// - Complexity: O(*n*)
-  @discardableResult
-  mutating func rotateSubrange(
-    _ bounds: Range<Index>, shiftingToStart middle: Index
-  ) -> Index
 }
 
 // In the stdlib, this conformance wouldn't be needed
 extension Array : MutableCollectionAlgorithms {  }
+extension ArraySlice : MutableCollectionAlgorithms {  }
```

Comment:
Do you want to add conformances for all the `MutableXXXSlice` types too?

Having to do this seems like a shame; it's making me want to add an empty `_AnyCollection`, `_AnyBidirectionalCollection`, and `_AnyRandomAccessCollection` protocol hierarchy to the standard library just to make this work, so we could then use `is` and manually dispatch to the right implementation for different traversal categories, since at this level, the requirements are all present on `Collection`.

Reply:
Added!